### PR TITLE
[BUG] Add overwrite to export_raw for EDF conversions

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Fixes: #929 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/tests/test_write.py
Comment (adam2392) [inline mne_bids/tests/test_write.py:2891]: These if/else statements are annoying to catch... Is there a way we can just ignore all warnings in a unit test?

Comment (hoechenberger) [reply mne_bids/tests/test_write.py:2891]: Yes, but we want to be as specific as we can be. So I think we'll have to live with this…

But these lines after `# should work when writing the second time` should be dedented one level, no?!

Comment (adam2392) [reply mne_bids/tests/test_write.py:2891]: Good point. 

I kept them within the pytest warning catch because I think those warnings will still fire off no matter what.

Comment (hoechenberger) [reply mne_bids/tests/test_write.py:2891]: Then I don't understand why we're even calling the function twice 🤔

**Review by sappelhoff — APPROVED** [review]
+1 to merge when CIs are green.

Thanks Adam!

**Review by hoechenberger — CHANGES_REQUESTED** [review]
I'll want have a look at the tests before we merge, will get back to you shortly.

File: mne_bids/tests/test_write.py
Comment (hoechenberger) [inline mne_bids/tests/test_write.py:2881]: Can you explain what that is supposed to achieve?

I dedented these lines and then discovered the same warning is still being thrown. So I'm wondering why you're even testing this?

File: mne_bids/tests/test_write.py
Comment (hoechenberger) [inline mne_bids/tests/test_write.py:2861]: Can you add a tiny docstring here, explaining what is being tested?

File: mne_bids/tests/test_write.py
Comment (adam2392) [reply mne_bids/tests/test_write.py:2881]: The first write_raw_bids gets the files there. The second one tests that overwrite=True works. If you remove the changed lines in `_export_edf()`, then this test will fail.

I'll add some details to the doc string

File: mne_bids/tests/test_write.py
Comment (adam2392) [reply mne_bids/tests/test_write.py]: Done.

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
@hoechenberger merge if happy

🙏 @adam2392

**Review by hoechenberger — APPROVED** [review]
Pushed a tiny change, good to merge once green

**Comment by hoechenberger** [comment]
Thanks @adam2392!

**Pull request merged**
